sympy/stats/stochastic_process_types.py

```diff
@@ -971,37 +950,372 @@ def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
                          (1 - self.p, Eq(x, self.failure)),
                          (S.Zero, True))
+
+
+dict_rv_subs = {'Poisson': lambda rv: Poisson(rv.name, lamda=rv.pspace.process.lamda*rv.key),
+                'Bernoulli': lambda rv: Bernoulli(rv.name, p=rv.pspace.process.p,
+                            succ=rv.pspace.process.success, fail=rv.pspace.process.failure)
+               }
+
+
+def _rvindexed_subs(process, expr, condition=None):
+    r"""
+    Substitutes the RandomIndexedSymbol with the RandomSymbol with
+    same name, distribution and probability as RandomIndexedSymbol.
+
+    Parameters
+    ==========
+
+    process : str
+       Key value of dict_rv_subs RandomVariable used to substitute
+    expr: RandomIndexedSymbol, Relational, Logic
+        Condition for which expectation has to be computed. Must
+        contain a RandomIndexedSymbol of the process.
+    condition: Relational, Logic
+        The given conditions under which computations should be done.
+
+    """
+
+    rvs_expr = random_symbols(expr)
+    if len(rvs_expr) != 0:
+        swapdict_expr = {}
+        for rv in rvs_expr:
+            if isinstance(rv, RandomIndexedSymbol):
+                newrv = dict_rv_subs[process](rv)
+                swapdict_expr[rv] = newrv
+        expr = expr.subs(swapdict_expr)
+    rvs_cond = random_symbols(condition)
+    if len(rvs_cond)!=0:
+        swapdict_cond = {}
+        for rv in rvs_cond:
+            if isinstance(rv, RandomIndexedSymbol):
+                newrv = dict_rv_subs[process](rv)
+                swapdict_cond[rv] = newrv
+        condition = condition.subs(swapdict_cond)
+    return expr, condition
+
+
+def _expectation_stoch(process, expr, condition=None, evaluate=True, **kwargs):
+    r"""
+    Internal method for computing expectation of indexed RV.
+
+    Parameters
+    ==========
+
+
+    process : str
+       Key value of dict_rv_subs RandomVariable used to substitute
+    expr: RandomIndexedSymbol, Relational, Logic
+        Condition for which expectation has to be computed. Must
+        contain a RandomIndexedSymbol of the process.
+    condition: Relational, Logic
+        The given conditions under which computations should be done.
+
+    Returns
+    =======
+
+    Expectation of the RandomIndexedSymbol.
+
+    """
+    new_expr, new_condition = _rvindexed_subs(process, expr, condition)
+
+    if not is_random(new_expr):
+        return new_expr
+    new_pspace = pspace(new_expr)
+    if new_condition is not None:
+        new_expr = given(new_expr, new_condition)
+    if new_expr.is_Add:  # As E is Linear
+        return Add(*[new_pspace.compute_expectation(
+                    expr=arg, evaluate=evaluate, **kwargs)
+                    for arg in new_expr.args])
+    return new_pspace.compute_expectation(
+            new_expr, evaluate=evaluate, **kwargs)
+
+def _probability_stoch(process, condition, given_condition=None, evaluate=True, **kwargs):
```

Comment:
You can shift this method to a super class of `BernoulliProcess` and `PoissonProcess` as static methods.

Reply:
This can be needed for DTSP, so, should be shifted to Stochatic Process class? I think it can be outside the class.